kernel/h1/src/timels.rs

```diff
@@ -74,46 +74,51 @@ impl Frequency for Freq256Khz {
 
 impl time::Time for Timels {
     type Frequency = Freq256Khz;
+    type Ticks = time::Ticks32;
 
-    fn now(&self) -> u32 {
+    fn now(&self) -> Self::Ticks {
         let regs = unsafe { &*self.registers };
         let cur = regs.value.get();
         let reload = regs.reload.get();
         let elapsed = reload - cur;
-        self.now.get().wrapping_add(elapsed)
-    }
-
-    fn max_tics(&self) -> u32 {
-        u32::max_value()
+        self.now.get().wrapping_add(elapsed).into()
     }
 }
     
 impl Alarm<'static> for Timels {
-    fn set_alarm(&self, tics: u32) {
-        let distance = tics.wrapping_sub(self.now.get());
+    fn set_alarm(&self, _reference: Self::Ticks, dt: Self::Ticks) {
+        // TODO: This is over-simplified, as it does not account for time since
+        // reference.
```

Comment:
IIRC you copied this TODO comment from my prototype, but you should address it. I think this logic needs to be fixed.

Reply:
Oops, yeah. It ended up doing the right thing, but that was by chance due to how the capsule is implemented. I fixed it and retested.